### PR TITLE
Update the build

### PR DESCRIPTION
**Pull request by pointlessone**
* The binary is built inside the container

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,17 +7,36 @@ govet examines Go source code and reports suspicious constructs, such as Printf
 ### Installation
 
 1. If you haven't already, [install the Code Climate CLI](https://github.com/codeclimate/codeclimate).
-2. Run `codeclimate engines:enable govet`. This command both installs the engine and enables it in your `.codeclimate.yml` file.
-3. You're ready to analyze! Browse into your project's folder and run `codeclimate analyze`.
+2. Add the following to yout Code Climate config:
```
Comment (wfleming) [inline README.md]: typo: "yout" -> "your"

File: README.md [hunk 2]
```diff
@@ -7,17 +7,36 @@ govet examines Go source code and reports suspicious constructs, such as Printf
...
+  ```yaml
+  engines:
```
Comment (wfleming) [inline README.md]: Please use the new v2 schema for examples now, which uses the `plugins:` key instead of `engines:`.

File: bin/build [hunk 3]
```diff
@@ -2,7 +2,7 @@
 set -ex
```
Comment (wfleming) [inline bin/build:2]: Is this script needed/used at all anymore? Doesn't look like it to me. Can it be removed? You also made changes here to have it refer to `gofmt`, which seems wrong regardless.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,14 @@
+.PHONY: image
```
Comment (wfleming) [inline Makefile:1]: `update` should also be in the `.PHONY` list.

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+IMAGE_NAME ?= codeclimate/codeclimate-govet
+
+update:
+	docker run \
+    --rm --interactive \
```
Comment (wfleming) [inline Makefile]: Indentation is off on these lines: should be a hard tab followed by two soft spaces so it's indented a bit further than `docker`.

File: bin/update [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
```
Comment (wfleming) [inline bin/update]: We should probably add `set -e` here to catch any failures.

File: bin/build (hunk 3 above)
Comment (pointlessone) [reply bin/build:2]: No, it's not. Good spot.

**Comment by pointlessone** [comment]
Updated

File: bin/build (hunk 3 above)
Comment (wfleming) [reply bin/build:2]: Looks like this file still hasn't been deleted?

File: bin/update [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+
+set -e
+
+apk add --no-cache jq
+gofmt_version=$(go version | cut -d ' ' -f 3)
```
Comment (wfleming) [inline bin/update]: This script is referring to `gofmt` still.

File: README.md [hunk 8]
```diff
@@ -7,17 +7,36 @@ govet examines Go source code and reports suspicious constructs, such as Printf
...
 This will build a `codeclimate/codeclimate-govet` image locally
 
+### Updating
+
+`govet` is a part of the Go distribution and shares version with it. Once in a
+while a new version of Go gets packaged and in order to get the latest version
+and force a new docker image build, we have to update the `engine.json` file.
+
+```console
+make update
```
Comment (wfleming) [inline README.md]: There's a presumption in these instructions that you have the latest `golang:alpine` image where you run `make update`, but I don't think that's reliable. If you had built an older version of this library on your laptop & then a new go version was released you'd probably still have the older `golang:alpine` on your laptop when you got around to this unless you'd explicitly pulled it. Is it worth adding a `docker pull golang:alpine` to these steps?

**Review by wfleming — APPROVED** [review]

**Pull request merged**